[BE] typing for decorators - fx/_compatibility

aorenste opened the pull request:
Summary: See #131429

Test Plan: unit tests pass

Differential Revision: D61493706


cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @rec

Skylion007 commented on torch/_inductor/pattern_matcher.py:
```diff
@@ -1124,7 +1124,7 @@ def replace(
                     return
 
-                new = typing.cast(Sequence[torch.fx.Node], new)
+                new = typing.cast(Sequence[torch.fx.Node], new)  # type: ignore[redundant-cast]
```
Wait, this typing.cast should just be removed

aorenste committed:
[BE] typing for decorators - fx/_compatibility (#134054)

Summary:
X-link: https://github.com/ctrl-labs/src2/pull/33884

X-link: https://github.com/pytorch/executorch/pull/4810

X-link: https://github.com/pytorch/torchrec/pull/2322

Pull Request resolved: https://github.com/pytorch/pytorch/pull/134054

See #131429

Test Plan: unit tests pass

Reviewed By: laithsakka

Differential Revision: D61493706
